unikraft/app/application.go

```diff
@@ -893,3 +896,48 @@ func (app application) Save(ctx context.Context) error {
 func (app application) Volumes() []*volume.VolumeConfig {
 	return app.volumes
 }
+
+func (app application) RemoveLibrary(ctx context.Context, libraryName string) error {
+	isLibraryExistInProject := false
+	for libKey, lib := range app.libraries {
+		if lib.Name() == libraryName {
+			isLibraryExistInProject = true
+			delete(app.libraries, libKey)
+
+			yamlFile, err := os.ReadFile(app.kraftfile.path)
+			if err != nil {
+				return err
+			}
+
+			yamlMap := app
+			err = yaml.Unmarshal(yamlFile, &yamlMap)
+			if err != nil {
+				return err
+			}
+
+			delete(yamlMap.libraries, libKey)
+			byteData, err := yaml.Marshal(yamlMap)
```

Comment:
This method breaks the ordering of the the `kraft.yaml`. It basically orders the keys alphabetically.

I suggest you try using `mergeOverExistingKraftfile(ctx, yamlMap)` and see if that helps. Also try `saveNewKraftfile(ctx, yamlMap)`. If that also does not work, look over this <https://stackoverflow.com/questions/70777144/go-unmarshal-yaml-and-preserve-order>

Reply:
`saveNewKraftfile(ctx, yamlMap)` resolve this problem :)